folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -188,14 +190,16 @@ private DefaultRawMemcacheClient(
     this.address = address;
     this.executor = executor;
     this.timeoutMillis = timeoutMillis;
+    this.metrics = metrics;
     this.maxSetLength = maxSetLength;
     this.channel = checkNotNull(channel, "channel");
     this.flusher = new BatchFlusher(channel, batchSize);
     this.outstandingRequestLimit = outstandingRequestLimit;
+    this.pendingCounterLimit = Math.max(Integer.MAX_VALUE / 2, outstandingRequestLimit);
```

Comment:
See this code was moved from below, but still not sure why we use this max. If you know, maybe add a comment on it?

Reply:
I could explain here... but better to add comment - will do!